http/analysis-service/server.py

```diff
@@ -613,36 +670,36 @@ def edit_scheduled_job(job_id):
 
     if request.method == 'GET':
         # Show job details.
-        return render_template("schedule.html", values=job_to_form(old_job))
+        return render_template(template, values=job_to_form(old_job))
     # else: request.method == 'POST'
     # update this job's details
     if request.form['job-id'] != job_id:
         return "Mismatched job id", 400
 
-    new_job, job_meta, errors = validate_job_form(request, is_update=True, old_job=old_job)
+    new_job, job_meta, errors = _validate_job_form(is_cluster, request, is_update=True, old_job=old_job)
 
     # If there were any errors, stop and re-display the form.
     if errors:
-        return render_template("schedule.html", values=request.form, errors=errors)
+        return render_template(template, values=request.form, errors=errors)
 
     # Upload code if need be:
-    if request.files['code-tarball']:
-        err = upload_code(request.form["job-name"], request.files["code-tarball"])
+    if request.files['code']:
+        err = upload_code(request.form["job-name"], request.files["code"])
         if err is not None:
-            errors["code-tarball"] = err
-            return render_template("schedule.html", values=request.form, errors=errors)
+            errors["code"] = err
+            return render_template(template, values=request.form, errors=errors)
 
     result = update_job(new_job)
 
     if result.rowcount > 0:
         print "Updated job id", job_id
         jobs = []
-        for j in get_jobs():
+        for j in get_jobs(is_cluster=is_cluster):
```

Comment:
It looks like this will overwrite the crontab schedule with either the cluster jobs or non-cluster jobs only (it should include *all* jobs when updating crontab).